lib/galaxy/workflow/modules.py

```diff
@@ -1911,7 +1931,7 @@ def execute(self, trans, progress, invocation_step, use_cached_job=False):
 
         param_combinations = []
         if collection_info:
-            iteration_elements_iter = collection_info.slice_collections()
+            iteration_elements_iter = list(collection_info.slice_collections())
```

Comment:
```suggestion
            iteration_elements_iter = collection_info.slice_collections()
```
can be eager, that was just for debugging. Just want to let the tests run to completion before doing this.